### PR TITLE
Added methods in QuestionController: CreateGrammar,CreateAudition

### PR DESCRIPTION
**Pull request by ElenaBurak**
Added GrammarQuestionDTO,  AudioQuestionDTO
Changed QuestionProfile,QuestionService
Add Migration DeletedFieldsInTableQuestion

### CODE REVIEW

File: exalearn-api/ExaLearn.WebApi/appsettings.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
   },
   "AllowedHosts": "*",
   "ConnectionStrings": {
-    "DbContext": "Server=localhost;Port=5432;Username=postgres;Password=;Database=exalern_db",
+    "DbContext": "Server=localhost;Password=123456;Port=5432;Username=postgres;Password=123;Database=exalern_db",
     "HostingContext": "linkToFilesHosting"
```
Comment (artemgael96) [inline exalearn-api/ExaLearn.WebApi/appsettings.json]: revert your private changes for DbContext

File: exalearn-api/ExaLearn.Bl/DTO/AudioQuestionDTO.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+{
+    public class AudioQuestionDTO
+    {
+        public LevelType Level { get; set; }
+
+        public string FileId { get; set; }
+
+        public string Question { get; set; }
+
+        public List<AnswerDTO> AnswersId { get; set; }
```
Comment (artemgael96) [inline exalearn-api/ExaLearn.Bl/DTO/AudioQuestionDTO.cs]: why are you using AnswersId? maybe change to AnswerId?

File: exalearn-api/ExaLearn.Bl/DTO/GrammarQuestionDTO.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+namespace ExaLearn.Bl.DTO
+{
+    public class GrammarQuestionDTO
+    {   
+        public LevelType Level { get; set; }
+
+        public string Question { get; set; }
+
+        public List<AnswerDTO> AnswersId { get; set; }
```
Comment (artemgael96) [inline exalearn-api/ExaLearn.Bl/DTO/GrammarQuestionDTO.cs]: the same

File: exalearn-api/ExaLearn.Bl/DTO/QuestionDTO.cs [hunk 4]
```diff
@@ -1,17 +1,20 @@
...
+        public AudioFileDTO AudioFile { get; set; }
 
         public QuestionType Type { get; set; }
 
         public string Text { get; set; }
 
-        public string Description { get; set; }
-
-        public int Score { get; set; }
+        public List<AnswerDTO> Answers { get; set; }
```
Comment (artemgael96) [inline exalearn-api/ExaLearn.Bl/DTO/QuestionDTO.cs:18]: the same

File: exalearn-api/ExaLearn.Bl/DTO/AudioQuestionDTO.cs (hunk 2 above)
Comment (romanstat) [reply exalearn-api/ExaLearn.Bl/DTO/AudioQuestionDTO.cs]: you have list AnswerDTO, but not List<int>, it is not ids, rename to Answers

File: exalearn-api/ExaLearn.Bl/DTO/GrammarQuestionDTO.cs (hunk 3 above)
Comment (romanstat) [reply exalearn-api/ExaLearn.Bl/DTO/GrammarQuestionDTO.cs]: Answers

File: exalearn-api/ExaLearn.Bl/Interfaces/IQuestionService.cs [hunk 5]
```diff
@@ -7,12 +7,14 @@ namespace ExaLearn.Bl.Interfaces
 {
     public interface IQuestionService
     {
-        public Task<QuestionDTO> CreateAsync(QuestionDTO question);
+        public Task<GrammarQuestionDTO> CreateGrammarQuestionAsync(GrammarQuestionDTO question);
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Bl/Interfaces/IQuestionService.cs:10]: CreateTopic for essay and speaking?

File: exalearn-api/ExaLearn.WebApi/appsettings.json (hunk 1 above)
Comment (romanstat) [reply exalearn-api/ExaLearn.WebApi/appsettings.json]: +

File: exalearn-api/ExaLearn.Bl/DTO/AudioQuestionDTO.cs [hunk 6]
```diff
@@ -0,0 +1,16 @@
+﻿using Shared.Enums;
+using System.Collections.Generic;
+
+namespace ExaLearn.Bl.DTO
+{
+    public class AudioQuestionDTO
+    {
+        public LevelType Level { get; set; }
+
+        public string FileId { get; set; }
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Bl/DTO/AudioQuestionDTO.cs]: Id is string type? maybe it is url

File: exalearn-api/ExaLearn.Bl/DTO/AudioQuestionDTO.cs [hunk 7]
```diff
@@ -0,0 +1,17 @@
+﻿using Shared.Enums;
+using System.Collections.Generic;
+using System.Security.Policy;
+
+namespace ExaLearn.Bl.DTO
+{
+    public class AudioQuestionDTO
+    {
+        public LevelType Level { get; set; }
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Bl/DTO/AudioQuestionDTO.cs:10]: add int Id

File: exalearn-api/ExaLearn.Bl/DTO/GrammarQuestionDTO.cs [hunk 8]
```diff
@@ -0,0 +1,14 @@
+﻿using Shared.Enums;
+using System.Collections.Generic;
+
+namespace ExaLearn.Bl.DTO
+{
+    public class GrammarQuestionDTO
+    {   
+        public LevelType Level { get; set; }
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Bl/DTO/GrammarQuestionDTO.cs:10]: add int id

File: exalearn-api/ExaLearn.Bl/DTO/TopicQuestionDTO.cs [hunk 9]
```diff
@@ -0,0 +1,10 @@
+﻿using Shared.Enums;
+
+namespace ExaLearn.Bl.DTO
+{
+    public class TopicQuestionDTO
+    {
+        public LevelType Level { get; set; }
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Bl/DTO/TopicQuestionDTO.cs:9]: space

File: exalearn-api/ExaLearn.Bl/Mapping/QuestionProfile.cs [hunk 10]
```diff
@@ -9,6 +9,24 @@ public class QuestionProfile : Profile
...
+                .ForMember(x => x.Level, map => map.MapFrom(source => source.LevelType))
+                .ForMember(x => x.FileUrl, map => map.MapFrom(source => source.AudioFile.Url))
+                .ForMember(x => x.Question, map => map.MapFrom(source => source.Text))
+                .ForMember(x => x.Answers, map => map.MapFrom(source => source.Answers))
+                .ReverseMap();
+
+            CreateMap<Question, TopicQuestionDTO>()
+                .ForMember(x => x.Level, map => map.MapFrom(source => source.Type))
+                .ForMember(x => x.Topic, map => map.MapFrom(source => source.Text))               
+                .ReverseMap();
```
Comment (artemgael96) [inline exalearn-api/ExaLearn.Bl/Mapping/QuestionProfile.cs]: add for all create map .ForAllOtherMembers(x => x.Ignore())

File: exalearn-api/ExaLearn.Bl/Mapping/QuestionProfile.cs [hunk 11]
```diff
@@ -9,6 +9,25 @@ public class QuestionProfile : Profile
...
+                .ForMember(x => x.FileUrl, map => map.MapFrom(source => source.AudioFile.Url))
+                .ForMember(x => x.Question, map => map.MapFrom(source => source.Text))
+                .ForMember(x => x.Answers, map => map.MapFrom(source => source.Answers))
+                .ReverseMap();
+
+            CreateMap<Question, TopicQuestionDTO>()
+                .ForMember(x => x.Level, map => map.MapFrom(source => source.Type))
+                .ForMember(x => x.Topic, map => map.MapFrom(source => source.Text))
+                .ReverseMap()
+                .ForAllOtherMembers(x => x.Ignore());
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Bl/Mapping/QuestionProfile.cs]: add .ForAllOtherMembers(x => x.Ignore()); for all CreateMap

File: exalearn-api/ExaLearn.WebApi/Controllers/QuestionController.cs [hunk 12]
```diff
@@ -19,6 +19,24 @@ public QuestionController(IQuestionService questionService)
             _questionService = questionService;
         }
 
+        [HttpPost("createGrammar")]
```
Comment (romanstat) [inline exalearn-api/ExaLearn.WebApi/Controllers/QuestionController.cs:22]: try to add from swagger 3 type of question

Comment (ElenaBurak) [reply exalearn-api/ExaLearn.WebApi/Controllers/QuestionController.cs:22]: added

File: exalearn-api/ExaLearn.Bl/DTO/TopicQuestionDTO.cs [hunk 13]
```diff
@@ -0,0 +1,13 @@
+﻿using Shared.Enums;
+
+namespace ExaLearn.Bl.DTO
+{
+    public class TopicQuestionDTO
+    {
+        public int Id { get; set; }
+
+        public LevelType Level { get; set; }
+
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Bl/DTO/TopicQuestionDTO.cs:10]: add QuestionType, front sent topic for essay or speaking

**Review by romanstat — APPROVED** [review]

**Review by vlad12345123 — APPROVED** [review]

**Pull request merged**